src/openforms/submissions/models/submission_value_variable.py

```diff
@@ -358,28 +358,24 @@ def to_python(self) -> Any:
                 return aware_date.date()
 
             maybe_naive_datetime = parse_datetime(self.value)
-            if maybe_naive_datetime is not None:
-                if timezone.is_aware(maybe_naive_datetime):
-                    return maybe_naive_datetime.date()
-                return timezone.make_aware(maybe_naive_datetime).date()
+            if maybe_naive_datetime is None:
+                return ""
 
-            raise ValueError(f"Could not parse date '{self.value}'")  # pragma: nocover
+            if timezone.is_aware(maybe_naive_datetime):
+                return maybe_naive_datetime.date()
+            return timezone.make_aware(maybe_naive_datetime).date()
```

Comment:
I just moved this line, but there was no test for it. Although I think this line may be unreacheable, because if a field of type "date" gets a datetime string in isoformat without timezone, then the `parse_date` on L355 raises an error.

Reply:
theory holds up with the coverage on master branch: https://app.codecov.io/gh/open-formulieren/open-forms/blob/master/src%2Fopenforms%2Fsubmissions%2Fmodels%2Fsubmission_value_variable.py#L364